presto-main/src/main/java/com/facebook/presto/operator/window/AggregateWindowFunction.java

```diff
@@ -80,7 +79,7 @@ else if ((frameStart == currentStart) && (frameEnd >= currentEnd)) {
     private void accumulate(int start, int end)
     {
         if (function.getParameterTypes().size() == 0) {
-            accumulator.addInput(new Page(end - start + 1));
+            accumulator.addInput(end - start + 1);
```

Comment:
You're right. In current implementation we don't need special case for no argument functions. Previously (before the simplification you suggested) it was needed because of the complicated `isNull` strategy and iteration inside of `addInput(Index)`.
Removed that bit. Although, I added as a separate commit a missing test which was previously introduced by that (already removed) commit. It's 2db34610540fc1941b3080fb5d539e4b359b74e0.
Fixup for the 8491cbab5740ecd822d5c7e0c741f30d3ae15cff is here: c20ada83e9e775426a4b5f7c1b9b8028903ca0d0. It's to prevent building unnecessary page when there're no parameters to the function.